### ISSUE TITLE
Unrelated Liquibase error when using jooq-meta-extensions with Spring Boot

### ISSUE THREAD

**Issue by cemremengu** [jOOQ/jOOQ#10268]
The examples does not have liquibase dependencies but for some reason once I add JOOQ into the system I start to get the error below. Is there a configuration that I am missing?

```
<dependency>
	<groupId>org.springframework.boot</groupId>
	<artifactId>spring-boot-starter-jooq</artifactId>
	<!-- Exclude the jOOQ Open Source Edition -->
	<exclusions>
		<exclusion>
			<groupId>org.jooq</groupId>
			<artifactId>jooq</artifactId>
		</exclusion>
	</exclusions>
</dependency>
<dependency>
	<groupId>org.jooq.trial</groupId>
	<artifactId>jooq</artifactId>
	<version>3.13.2</version>
</dependency>
<dependency>
	<groupId>org.jooq.trial</groupId>
	<artifactId>jooq-meta-extensions</artifactId>
	<version>3.13.2</version>
</dependency>
<plugin>
	<groupId>org.jooq.trial</groupId>
	<artifactId>jooq-codegen-maven</artifactId>
	<version>3.13.2</version>
	<executions>
		<execution>
			<goals>
				<goal>generate</goal>
			</goals>
		</execution>
	</executions>
	<configuration>
		<generator>
			<database>
				<name>org.jooq.meta.extensions.jpa.JPADatabase</name>
				<properties>
					<property>
						<key>packages</key>
						<value>com.myapp.entity</value>
					</property>
					<property>
						<key>useAttributeConverters</key>
						<value>true</value>
					</property>
					<property>
						<key>unqualifiedSchema</key>
						<value>none</value>
					</property>
				</properties>
			</database>
		</generator>
	</configuration>
</plugin>
```

```
org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'liquibase' defined in class path resource [org/springframework/boot/autoconfigure/liquibase/LiquibaseAutoConfiguration$LiquibaseConfiguration.class]: Invocation of init method failed; nested exception is liquibase.exception.ChangeLogParseException: Error parsing classpath:/db/changelog/db.changelog-master.yaml
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1778) ~[spring-beans-5.1.10.RELEASE.jar:5.1.10.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:593) ~[spring-beans-5.1.10.RELEASE.jar:5.1.10.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:515) ~[spring-beans-5.1.10.RELEASE.jar:5.1.10.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:320) ~[spring-beans-5.1.10.RELEASE.jar:5.1.10.RELEASE]
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:222) ~[spring-beans-5.1.10.RELEASE.jar:5.1.10.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:318) ~[spring-beans-5.1.10.RELEASE.jar:5.1.10.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:199) ~[spring-beans-5.1.10.RELEASE.jar:5.1.10.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:307) ~[spring-beans-5.1.10.RELEASE.jar:5.1.10.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:199) ~[spring-beans-5.1.10.RELEASE.jar:5.1.10.RELEASE]
	at org.springframework.context.support.AbstractApplicationContext.getBean(AbstractApplicationContext.java:1105) ~[spring-context-5.1.10.RELEASE.jar:5.1.10.RELEASE]
	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:867) ~[spring-context-5.1.10.RELEASE.jar:5.1.10.RELEASE]
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:549) ~[spring-context-5.1.10.RELEASE.jar:5.1.10.RELEASE]
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:141) ~[spring-boot-2.1.9.RELEASE.jar:2.1.9.RELEASE]
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:744) ~[spring-boot-2.1.9.RELEASE.jar:2.1.9.RELEASE]
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:391) ~[spring-boot-2.1.9.RELEASE.jar:2.1.9.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:312) ~[spring-boot-2.1.9.RELEASE.jar:2.1.9.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1215) ~[spring-boot-2.1.9.RELEASE.jar:2.1.9.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1204) ~[spring-boot-2.1.9.RELEASE.jar:2.1.9.RELEASE]
	at com.ttgint.cmex.CmexApplication.main(CmexApplication.java:15) ~[classes/:na]
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:na]
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:na]
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:na]
	at java.base/java.lang.reflect.Method.invoke(Method.java:566) ~[na:na]
	at org.springframework.boot.devtools.restart.RestartLauncher.run(RestartLauncher.java:49) ~[spring-boot-devtools-2.1.9.RELEASE.jar:2.1.9.RELEASE]
Caused by: liquibase.exception.ChangeLogParseException: Error parsing classpath:/db/changelog/db.changelog-master.yaml
	at liquibase.parser.core.yaml.YamlChangeLogParser.parse(YamlChangeLogParser.java:84) ~[liquibase-core-3.6.3.jar:na]
	at liquibase.Liquibase.getDatabaseChangeLog(Liquibase.java:217) ~[liquibase-core-3.6.3.jar:na]
	at liquibase.Liquibase.update(Liquibase.java:190) ~[liquibase-core-3.6.3.jar:na]
	at liquibase.Liquibase.update(Liquibase.java:179) ~[liquibase-core-3.6.3.jar:na]
	at liquibase.integration.spring.SpringLiquibase.performUpdate(SpringLiquibase.java:353) ~[liquibase-core-3.6.3.jar:na]
	at liquibase.integration.spring.SpringLiquibase.afterPropertiesSet(SpringLiquibase.java:305) ~[liquibase-core-3.6.3.jar:na]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1837) ~[spring-beans-5.1.10.RELEASE.jar:5.1.10.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1774) ~[spring-beans-5.1.10.RELEASE.jar:5.1.10.RELEASE]
	... 23 common frames omitted
Caused by: java.io.FileNotFoundException: class path resource [db/changelog/db.changelog-master.yaml] cannot be resolved to URL because it does not exist
	at org.springframework.core.io.ClassPathResource.getURL(ClassPathResource.java:195) ~[spring-core-5.1.10.RELEASE.jar:5.1.10.RELEASE]
	at liquibase.integration.spring.SpringLiquibase$SpringResourceOpener.getResourcesAsStream(SpringLiquibase.java:599) ~[liquibase-core-3.6.3.jar:na]
	at liquibase.util.StreamUtil.singleInputStream(StreamUtil.java:186) ~[liquibase-core-3.6.3.jar:na]
	at liquibase.parser.core.yaml.YamlChangeLogParser.parse(YamlChangeLogParser.java:27) ~[liquibase-core-3.6.3.jar:na]
	... 30 common frames omitted
```

**Comment by lukaseder** [comment]
Thanks for your message. No, there's no such requirement. `jooq-meta-extensions` has a dependency on Liquibase for those users who need it, but this shouldn't activate it automatically in Spring Boot. Might be a Spring Boot bug? How can this be reproduced in a most minimal way? If you have some time, would it be possible to create an MCVE? https://github.com/jOOQ/jOOQ-mcve

**Comment by cemremengu** [comment]
Sure here it is: https://github.com/cemremengu/jooq-spring-jpa-mcve

**Comment by cemremengu** [comment]
If I exclude the dependency as below, it works:

```
    <dependency>
      <groupId>org.jooq.trial</groupId>
      <artifactId>jooq-meta-extensions</artifactId>
      <version>${jooq.version}</version>
      <exclusions>
        <exclusion>
          <groupId>org.liquibase</groupId>
          <artifactId>liquibase-core</artifactId>
        </exclusion>
      </exclusions>
    </dependency>
```

**Comment by lukaseder** [comment]
Thanks a lot for this. I can reproduce the problem using your MCVE and will look into whether this needs to be fixed on the jOOQ or Spring Boot side.

**Comment by cemremengu** [comment]
Not sure if it is related but while it works with the dependency exclusion, it tries to excute ddl statements during startup (drop create table) on an unknown (I guess internal) H2 database and select queries seem to return empty result sets.

**Comment by lukaseder** [comment]
I can reproduce this in our own Spring Boot example by simply adding the `jooq-meta-extensions` dependency:
https://github.com/jOOQ/jOOQ/tree/main/jOOQ-examples/jOOQ-spring-boot-example

**Comment by lukaseder** [comment]
The way I see it, this is a problem related to Liquibase itself. For some reason, `SpringLiquibase` activates itself in this kind of setup, when it is not explicitly requested.

**Comment by lukaseder** [comment]
We could make our Liquibase (and Hibernate) dependencies:

```xml
<scope>provided</scope>
<optional>true</optional>
```

... which means that users have to provide the dependency themselves. Maybe there's a better solution, though. I always liked the fact that `jooq-meta-extensions` works out of the box.

**Comment by lukaseder** [comment]
Another workaround is to pass the `-Dliquibase.shouldRun=false` flag

**Comment by lukaseder** [comment]
I've created a bug in Spring Boot to see if something can be done on their side, or what they recommend we do: https://github.com/spring-projects/spring-boot/issues/22159

**Comment by lukaseder** [comment]
We could try to put the `shouldRun=false` property on the classpath for liquibase to pick up. That would solve our issue here, but would possibly turn off a desireable Liquibase feature for those users who actually *want* to use Liquibase.

**Comment by lukaseder** [comment]
It doesn't look like doing additional magic distinguishing between "accidental" and "wanted" dependencies is on Spring Boot's radar: https://github.com/spring-projects/spring-boot/issues/22159#issuecomment-651687125

So, the ideal solution, from what I can tell, and from the jOOQ side is to move the `LiquibaseDatabase` in yet another Maven module `jooq-meta-extensions-liquibase`. This will make the transitive dependency explicit, and this accident will no longer happen.

I might do the same thing for the Hibernate dependency, which may be unwanted if someone wants to use the `LiquibaseDatabase`

**Comment by lukaseder** [comment]
Will solve this via https://github.com/jOOQ/jOOQ/issues/10331

**Comment by lukaseder** [comment]
Fixed via #10331. The fix will not be backported as Maven modules are renamed backwards incompatibly. There are workarounds for the time being.

Thanks again for the detailed report!

**Comment by cemremengu** [comment]
Thanks for the hard work! I think this what you have decided is the best solution to this 👍
